### ISSUE TITLE
Results of the Valiation and Test set on Waymo Open Dataset

### ISSUE THREAD

**Issue by ziqipang** [tianweiy/CenterPoint#134]
Dear Tianwei, Hi!

Thank you for your excellent paper, which is both insightful and achieves outstanding performance!

I am wondering if you may release your results for the validation set and test set on Waymo Open Dataset? The format of .bin will work for me. 

Thank you!

Best Regards,
Ziqi

**Comment by tianweiy** [comment]
I can't find my old models so I will train a new one by this weekend. I will update you then. Please refer to the other repo for waymo setup. These two repos are not compatible at the moment.

**Comment by tianweiy** [comment]
Please send me an email for this.

**Comment by ziqipang** [comment]
Thank you! @tianweiy

**Comment by YoushaaMurhij** [comment]
@tianweiy, I tried to validate on waymo val-set and I got my_pred.bin. I want to submit it to the waymo server. The next step I see is to convert the results :
```bash
mkdir /tmp/my_model
metrics/tools/create_submission  --input_filenames='/tmp/preds.bin' --output_filename='/tmp/my_model/model' --submission_filename='metrics/tools/submission.txtpb
```
I am stuck here after cloning waymo devkit. I have only `create_submission.cc` in `metrics/tools.`
I am note sure how to run this command. Can you please tell me how to get the results?
Thanks!

**Comment by tianweiy** [comment]
you need to compile it with Bazel https://github.com/waymo-research/waymo-open-dataset/blob/master/docs/quick_start.md

**Comment by YoushaaMurhij** [comment]
For generating the test-set `preds.bin` I need to add --testset flag and specify the `"data/Waymo/infos_test_01sweeps_filter_zero_gt.pkl"` path in `test_anno `in the config.
Is there anything else?
I got this error:
```bash
  File "/home/trainer/centerpoint/det3d/datasets/waymo/waymo.py", line 62, in __len__
    self.load_infos(self._info_path)
  File "/home/trainer/centerpoint/det3d/datasets/waymo/waymo.py", line 52, in load_infos
    with open(self._info_path, "rb") as f:
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

**Comment by tianweiy** [comment]
That's all. You may need to specify the info path to data/Waymo/infos_test_01sweeps_filter_zero_gt.pkl in your config. Now it doesn't have the path (value is none)

**Comment by YoushaaMurhij** [comment]
Thanks.  My bad! I was using another config. It worked but I got this with second stage config:

```bash
File "tools/dist_test.py", line 201, in main
    result_dict, _ = dataset.evaluation(copy.deepcopy(predictions), output_dir=args.work_dir, testset=args.testset)
  File "/home/trainer/centerpoint/det3d/datasets/waymo/waymo.py", line 100, in evaluation
    _create_pd_detection(detections, infos, output_dir)
  File "/home/trainer/centerpoint/det3d/datasets/waymo/waymo_common.py", line 114, in _create_pd_detection
    f.write(objects.SerializeToString())
ValueError: Message waymo.open_dataset.Objects exceeds maximum protobuf size of 2GB: 2477912500
```

**Comment by tianweiy** [comment]
not sure which config / model you are using. Sometimes if the model's accuracy is not good, you will get too much false positives and exceed this 2GB limit. I remembered there are some way to bypass this limit by creating multiple result file but you need to search in waymo repo's issue section

**Comment by YoushaaMurhij** [comment]
Thanks for you suggestion! Your results on public WAYMO leader-board which is about 0.78 mAP are not your repo. Is it another private (internal) model that you use?

**Comment by tianweiy** [comment]
The 72 is my repo. The 78 is others(not my submission

**Comment by YoushaaMurhij** [comment]
I tried to add another shared layers in CenterHead but when I run second stage code it can't see those layers:

`
unexpected key in source state_dict: bbox_head.f_shared_conv.0.weight, bbox_head.f_shared_conv.0.bias, bbox_head.f_shared_conv.1.weight, bbox_head.f_shared_conv.1.bias, bbox_head.f_shared_conv.1.running_mean, bbox_head.f_shared_conv.1.running_var, bbox_head.f_shared_conv.1.num_batches_tracked`

Do I need to initialize them in another place?
Thanks!

**Comment by tianweiy** [comment]
when does this happen? This is normal if you get this message when it started loading weights from pretrained models (for two stage training) as the pretrained weight file doesn't get your new layers.

**Comment by YoushaaMurhij** [comment]
I added a shared layer in CenterPoint and trained 1st stage After that, I tried to load those weights in the configs of  2nd stage and got this issue.

**Comment by tianweiy** [comment]
Sounds a little weird, probably you also need to modify the 2nd stage config? You can just print out the model before loading weights to see why it is not there

**Comment by YoushaaMurhij** [comment]
I think I found the problem: I am using a modified version of CenterHead in 1st stage (another .py file) and when o try to train the second stage It assumes that the the 1st stage uses the original CenterHead  code (.py) without this shared layer that I added.
I think I need to specify which config I used in the 1st stage inside the config of the 2nd stage or something like that!
Can i do that directly? or I should rearrange the code?
